testing/go/dolt_functions_test.go

```diff
@@ -20,7 +20,2381 @@ import (
+				"CREATE TABLE t_default_simple (v1 INT8 DEFAULT 22, v2 INT8);",
+				"CREATE TABLE t_checked (v1 NUMERIC CHECK (v1 > 0 AND v1 <= 100));",
+				"CREATE TABLE t_fk_parent (pk INT4 PRIMARY KEY);",
+				"CREATE TABLE t_fk_child (pk INT4 REFERENCES t_fk_parent(pk), PRIMARY KEY(pk));",
+				"CREATE TABLE t_unique (v1 INT4 UNIQUE);",
+				"CREATE TABLE t_generated (v1 INT8, v2 INT8 GENERATED ALWAYS AS (v1 * 1000) STORED);",
+				"CREATE TABLE t_trigger (v1 INT8);",
+				"CREATE FUNCTION f_trigger() RETURNS TRIGGER AS $$ BEGIN NEW.v1 := NEW.v1 * 3; RETURN NEW; END; $$ LANGUAGE plpgsql;",
+				"CREATE TRIGGER trig_trigger BEFORE INSERT OR UPDATE ON t_trigger FOR EACH ROW EXECUTE FUNCTION f_trigger();",
+				"CREATE FUNCTION f_default() RETURNS INT8 AS $$ BEGIN RETURN 33; END; $$ LANGUAGE plpgsql;",
+				"CREATE TABLE t_default_func (v1 INT8 DEFAULT f_default(), v2 INT8);",
+				"INSERT INTO t_simple VALUES (1), (2), (3);",
+				"INSERT INTO t_composite VALUES (1, 100), (1, 101), (2, 100), (2, 101);",
+				"INSERT INTO t_array VALUES (ARRAY['abc']), (ARRAY['def','ghi']), (ARRAY['jkl','mno','pqr']);",
+				"INSERT INTO t_serial VALUES (DEFAULT), (DEFAULT), (DEFAULT);",
+				"INSERT INTO t_default_simple VALUES (DEFAULT, 5), (99, 6), (DEFAULT, 7);",
+				"INSERT INTO t_checked VALUES (1), (50), (100);",
+				"INSERT INTO t_fk_parent VALUES (10), (20), (30);",
+				"INSERT INTO t_fk_child VALUES (10), (20);",
+				"INSERT INTO t_unique VALUES (7), (8), (9);",
+				"INSERT INTO t_generated (v1) VALUES (1), (2), (10);",
+				"INSERT INTO t_trigger (v1) VALUES (5), (10), (0);",
+				"INSERT INTO t_default_func (v2) VALUES (1), (2);",
+				"SELECT length(DOLT_COMMIT('-A', '-m', 'initial')::text) = 34;",
+				"SELECT DOLT_BRANCH('original');",
+				"INSERT INTO t_simple VALUES (4);",
+				"INSERT INTO t_composite VALUES (3, 100);",
+				"INSERT INTO t_array VALUES (ARRAY['stu']);",
+				"INSERT INTO t_serial VALUES (DEFAULT);",
+				"INSERT INTO t_default_simple VALUES (98, 8);",
+				"INSERT INTO t_checked VALUES (99);",
+				"INSERT INTO t_fk_parent VALUES (40);",
+				"INSERT INTO t_fk_child VALUES (30);",
+				"INSERT INTO t_unique VALUES (10);",
+				"INSERT INTO t_generated (v1) VALUES (11);",
+				"CREATE OR REPLACE FUNCTION f_trigger() RETURNS TRIGGER AS $$ BEGIN NEW.v1 := NEW.v1 * 33; RETURN NEW; END; $$ LANGUAGE plpgsql;",
+				"INSERT INTO t_trigger (v1) VALUES (2);",
+				"CREATE OR REPLACE FUNCTION f_default() RETURNS INT8 AS $$ BEGIN RETURN 34; END; $$ LANGUAGE plpgsql;",
+				"INSERT INTO t_default_func (v2) VALUES (3);",
+				"SELECT length(DOLT_COMMIT('-A', '-m', 'initial')::text) = 34;",
+				"SELECT DOLT_CHECKOUT('original');",
+			},
+			Assertions: []ScriptTestAssertion{
+				{
+					Query:    "SELECT strpos(DOLT_MERGE('main')::text, '1,0,\"merge successful\"') > 1;",
+					Expected: []sql.Row{{"t"}},
+				},
+			},
+		},
+		{
+			Name: "--no-ff",
+			SetUpScript: []string{
+				"CREATE TABLE t_simple (pk INT4 PRIMARY KEY);",
```

Comment:
Might be worth putting these into a shared var so they could be reused between tests. Seems like these are copy/pasted a lot. Would also be helpful to comment on that shared var with the intent of why we are creating so many permutations – I don't think that will be immediately obvious to future maintainers reading these.